presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1278,19 +1278,16 @@ private void analyzeFiltersAndMasks(Table table, QualifiedObjectName name, Optio
                     .withRelationType(RelationId.anonymous(), new RelationType(fields))
                     .build();
 
-            Set<String> allColumns = fields.stream()
-                    .flatMap(field -> field.getName().stream())
-                    .collect(toImmutableSet());
-
-            Set<String> visibleColumns = filterColumns(name, allColumns);
-
             ImmutableMap.Builder<Field, List<ViewExpression>> columnMasks = ImmutableMap.builder();
             for (Field field : fields) {
-                if (field.getName().isPresent() && visibleColumns.contains(field.getName().orElseThrow())) {
+                if (field.getName().isPresent()) {
                     List<ViewExpression> masks = accessControl.getColumnMasks(session.toSecurityContext(), name, field.getName().get(), field.getType());
-                    columnMasks.put(field, masks);
 
-                    masks.forEach(mask -> analyzeColumnMask(session.getIdentity().getUser(), table, name, field, accessControlScope, mask));
+                    if (!masks.isEmpty() && checkCanSelectFromColumn(name, field.getName().orElseThrow())) {
```

Comment:
field.getName().orElseThrow() can be extracted to a variable

Reply:
@skrzypo987 Just to make sure I understand this change. What if there are no masks for a restricted column? Do we not check that like before? I guess my question is, will this change "leak" the restricted columns without masks.